pkg/wakaama/include/lwm2m_platform.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_wakaama
+ * @defgroup    lwm2m_platform Platform adaption for Wakaama package
+ * @brief Adaption of Wakaama LwM2M package to RIOT
+ * @{
+ * @file
+ * @brief Definitions and public API for Wakaama adaption layer
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+
+/**
+ * @defgroup lwm2m_platform_conf Wakaama LwM2M platform adaption compile configuration
+ * @ingroup config
+ * @{
+ */
+
+#ifndef LWM2M_PLATFORM_H
+#define LWM2M_PLATFORM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Size of allocation buffer */
+#ifndef LWM2M_TLSF_BUFFER
+#define LWM2M_TLSF_BUFFER     (10240 / sizeof(uint32_t))
```

Comment:
How did you come up with this size? I'm not suggesting it's wrong, just curious.

Reply:
The TLSF internal header uses a bit more than 3KiB, so the minimum here would be 4KiB. I took as a reference the current Lua buffer size, which is around 10KiB.

We could reduce this for now, but I think as more objects are introduced this will need to change.

Reply:
I am copy/pasting the TLSF [heap printer](https://github.com/RIOT-OS/RIOT/blob/00c21e2f8c97ecc99aed3c26c187917e3d4a1d22/sys/ps/ps.c#L149) into `lwm2m_platform.c` to ensure there are no leaks. I'll share this work, and let you know how much memory is used.

Reply:
I didn't see your comment before I pasted mine. :-) Thanks for that information. I'm not too worried about this default value. At the same time, I had to work with the overall system configuration to get it to fit into the 32 KiB of RAM on the samr21-xpro. Does it make sense to mention this parameter somewhere -- maybe the example README or Makefile?

Reply:
> Does it make sense to mention this parameter somewhere -- maybe the example README or Makefile?

The parameter is already in the Compile-time configuration group for Wakaama in doxygen, I will also mention it in the example README.

Reply:
Updated [my branch](https://github.com/kb2ma/RIOT/tree/pr/pkg/wakaama_rework) with a CLI command to dump TLSF buffer status. Feel free to pull this in. I think it will be helpful for memory tuning.

On samr21-xpro, around 160 bytes were used when starting Wakaama. I see 416 bytes used and 6572 free after executing `lwm2m start`. Memory use is stable across registrations and interaction with the Leshan UI. 

It will be interesting to see how these values change with a few SAUL sensors collecting and reporting data. I might shave a couple of KiB from the current example, but I wouldn't block the PR if you'd rather leave as is.

Reply:
Thanks! I added this and reduced the stack to 5KiB